packages/lib/isOutOfBounds.tsx

```diff
@@ -43,24 +189,91 @@ function isOutOfBounds(
     }
   }
 
-  switch (periodType) {
-    case PeriodType.ROLLING: {
-      const periodRollingEndDay = periodCountCalendarDays
-        ? dayjs().utcOffset(date.utcOffset()).add(periodDays, "days").endOf("day")
-        : dayjs().utcOffset(date.utcOffset()).businessDaysAdd(periodDays).endOf("day");
-      return date.endOf("day").isAfter(periodRollingEndDay);
-    }
+  return false;
+}
 
-    case PeriodType.RANGE: {
-      const periodRangeStartDay = dayjs(periodStartDate).utcOffset(date.utcOffset()).endOf("day");
-      const periodRangeEndDay = dayjs(periodEndDate).utcOffset(date.utcOffset()).endOf("day");
-      return date.endOf("day").isBefore(periodRangeStartDay) || date.endOf("day").isAfter(periodRangeEndDay);
-    }
+type PeriodLimits = {
+  rollingEndDay: dayjs.Dayjs | null;
+  rangeStartDay: dayjs.Dayjs | null;
+  rangeEndDay: dayjs.Dayjs | null;
+};
 
-    case PeriodType.UNLIMITED:
-    default:
-      return false;
+/**
+ * To be used when we work on just Dates(and not specific timeslots) to check boundaries
+ * e.g. It checks for Future Limits which operate on dates and not times.
+ */
+export function isDateOutOfBounds({
+  dateString,
+  periodLimits,
+}: {
+  dateString: dayjs.ConfigType;
+  periodLimits: PeriodLimits;
+  _skipRollingWindowCheck?: boolean;
+}) {
+  const log = logger.getSubLogger({ prefix: ["isDateOutOfBounds"] });
+
+  log.debug(
+    safeStringify({
+      dateString,
+      periodLimits: {
+        rollingEndDay: periodLimits.rollingEndDay?.format("YYYY-MM-DD"),
+        rangeStartDay: periodLimits.rangeStartDay?.format("YYYY-MM-DD"),
+        rangeEndDay: periodLimits.rangeEndDay?.format("YYYY-MM-DD"),
+      },
+    })
+  );
+
+  const endOfDay = dayjs(dateString);
+
+  if (periodLimits.rollingEndDay) {
+    const isAfterRollingEndDay = endOfDay.isAfter(periodLimits.rollingEndDay);
+    log.debug({
+      dateString,
+      endOfDay: endOfDay.format(),
+      isAfterRollingEndDay,
+      rollingEndDay: periodLimits.rollingEndDay.format(),
+    });
+    return isAfterRollingEndDay;
+  }
+
+  if (periodLimits.rangeStartDay && periodLimits.rangeEndDay) {
+    return endOfDay.isBefore(periodLimits.rangeStartDay) || endOfDay.isAfter(periodLimits.rangeEndDay);
   }
+  return false;
 }
 
-export default isOutOfBounds;
+export default function isOutOfBounds(
+  time: dayjs.ConfigType,
+  {
+    periodType,
+    periodDays,
+    periodCountCalendarDays,
+    periodStartDate,
+    periodEndDate,
+    utcOffset,
+  }: Pick<
+    EventType,
+    "periodType" | "periodDays" | "periodCountCalendarDays" | "periodStartDate" | "periodEndDate"
+  > & {
+    utcOffset: number;
+  },
+  minimumBookingNotice?: number
+) {
+  return (
+    isTimeOutOfBounds({ time, minimumBookingNotice }) &&
```

Comment:
if `isTimeOutOfBounds` is false, so minimum booking notice is respected, then `isDateOutOfBounds` is never called

So I am able to book a slot that doesn't respect the 'Limit future bookings' settings checked in `isDateOutOfBounds`

Reply:
Fixed this one as well and added a test.